src/Materials/Textures/Packer/packer.ts

```diff
@@ -0,0 +1,827 @@
+                return new Vector2(xStep * uvStep.x , yStep * uvStep.y);
+            break;
+        }
+
+        return Vector2.Zero();
+    }
+
+    /**
+    * Updates a Mesh to the frame data
+    * @param mesh that is the target
+    * @param frameID or the frame index
+    */
+    private _updateMeshUV(mesh: AbstractMesh, frameID: number): void {
+        let frame: TexturePackerFrame = (this.frames as any)[frameID];
+        let uvIn = mesh.getVerticesData(this.options.uvsIn || VertexBuffer.UVKind);
+        let uvOut = [];
+        let toCount = 0;
+
+        if (uvIn!.length) {
+            toCount = uvIn!.length || 0;
+        }
+
+        for (let i = 0; i < toCount; i += 2) {
+            uvOut.push(
+                ((uvIn as any)[i] * frame.scale.x) + frame.offset.x,
+                ((uvIn as any)[i + 1] * frame.scale.y) + frame.offset.y
+            );
+        }
+
+        mesh.setVerticesData(this.options.uvsOut || VertexBuffer.UVKind, uvOut);
+    }
+
+    /**
+    * Updates a Meshes materials to use the texture packer channels
+    * @param m is the mesh to target
+    * @param force all channels on the packer to be set.
+    */
+    private _updateTextureRefrences(m: AbstractMesh, force: boolean = false): void {
+        let mat = m.material;
+        let sKeys = Object.keys(this.sets);
+
+        let _dispose = (_t: any) => {
+             if ((_t.dispose)) {
+                _t.dispose();
+             }
+        };
+
+        for (let i = 0; i < sKeys.length; i++) {
+            let setName = sKeys[i];
+            if (!force) {
+                if (!mat) {
+                    return;
+                }
+                if ((mat as any)[setName] !== null) {
+                    _dispose((mat as any)[setName]);
+                    (mat as any)[setName] = (this.sets as any)[setName];
+                }
+            }else {
+                if ((mat as any)[setName] !== null) {
+                    _dispose((mat as any)[setName]);
+                }
+                (mat as any)[setName] = (this.sets as any)[setName];
+            }
+        }
+    }
+
+    /**
+    * Public method to set a Mesh to a frame
+    * @param m that is the target
+    * @param frameID or the frame index
+    * @param updateMaterial trigger for if the Meshes attached Material be updated?
+    */
+    public setMeshToFrame(m: AbstractMesh, frameID: number, updateMaterial: boolean = false): void {
+        this._updateMeshUV(m, frameID);
+        if (updateMaterial) {
+            this._updateTextureRefrences(m, true);
+        }
+    }
+
+    /**
+    * Starts the async promise to compile the texture packer.
+    * @param success callback for the load promise
+    * @param error callback for the load promise
+    */
+    public processAsync(success = (): void => {}, error = (err : string): void => {}): void {
+        setTimeout(() => {
+            this.promise = new Promise ((resolve, reject) => {

```

Comment:
Here is what I had in mind:

public processAsync(): Promise<void> {
    return new Promise ((resolve, reject) => {
...
})
}

Reply:
the user can then do:
```
pack.processAsync().then(() => {
// my stuff
}).catch(e => {
// handling error
});
```

Reply:
so then they do what like
```
let packPromise = pack.processAsync()

packPromise.then(success, error)
```
?

Reply:
and then I can just drop the this.promise

Reply:
packPromise.then(() => success()).catch(() => error())

Reply:
I did not see your comment my bad!

Reply:
I did not see your comment my bad!